packages/jsii-rosetta/lib/jsii/assemblies.ts

```diff
@@ -228,14 +234,11 @@ export async function allTypeScriptSnippets(
 
 /**
  * Replaces the file where the original assembly file *should* be found with a new assembly file.
+ * Detects whether or not there is a compressed assembly, and if there is, compresses the new assembly also.
  * Recalculates the fingerprint of the assembly to avoid tampering detection.
  */
-export function replaceAssembly(
-  assembly: spec.Assembly,
-  directory: string,
-  { compress = false }: { compress?: boolean } = {},
-) {
-  writeAssembly(directory, _fingerprint(assembly), { compress });
+export function replaceAssembly(assembly: spec.Assembly, directory: string) {
```

Comment:
This is a breaking change :(. If we must add the `{ compress = false }: { compress?: boolean } = {}` option in again, then fine. I really don't want to though, because in my mind, this was a bit of a stealth mode release and its unlikely that this revert actually breaks any users. Especially because the prior behavior was incorrect, and I would simply ignore the property if we have to keep it in...

Reply:
You could also just change the default behavior for `compress` -- JavaScript does not require default values are constants/literals, it can be a function call result...